src/chat_do.py

```
@@ -0,0 +1,406 @@
+        except Exception as exc:
+            print(f"[ChatDO._ensure_table] error={exc!r}")
+
+    async def _persist_message(self, msg_id: str, classroom_id: str, user_id: str, display_name: str, text: str) -> bool:
+        try:
+            raw_key = getattr(self.env, "CHAT_ENCRYPTION_KEY", None) or getattr(self.env, "ENCRYPTION_KEY", None)
+            enc_key = raw_key.strip() if isinstance(raw_key, str) and raw_key.strip() else ""
+            stored_content = await encrypt_aes(text, enc_key) if enc_key else text
+            await self.env.DB.prepare(
+                "INSERT INTO chat_message (id, classroom_id, user_id, display_name, content, created_at) VALUES (?, ?, ?, ?, ?, datetime('now'))"
+            ).bind(msg_id, classroom_id, user_id, display_name, stored_content).run()
+            return True
```

Ruff (0.16.0)

[warning] Consider moving this statement to an `else` block

(TRY300)